src/Controller/ProgramController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace App\Controller;
+
+use App\Repository\ProgramRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use App\Entity\Program;
+
+class ProgramController extends AbstractController
+{
+    /**
+     * Returns list of programs.
+     *
+     * @return JsonResponse
+     *
+     * @Route(
+     *     "/api/v1/programs",
+     *     name="programs",
+     *     methods={"GET"}
+     * )
+     */
+    public function getPrograms()
+    {
+        $programs = $this->getDoctrine()
+            ->getRepository(Program::class)
+            ->findAll();
+
+        if ($programs === null) {
+            return new JsonResponse("We could not find any programs", Response::HTTP_NOT_FOUND);
+        }
+
+        $programsArray = array();
+
+        foreach ($programs as $program) {
+            $programsArray[] = array(
+                'program_id' => $program->getId(),
+                'program_name' => $program->getProgramName(),
+                'program_url' => $program->getProgramUrl(),
+                'program_price' => $program->getProgramPrice(),
+            );
+        }
+
+        return new JsonResponse(
+            $programsArray,
+            JsonResponse::HTTP_OK
+        );
+    }
+
+    /**
+     * Returns program selected by the id.
+     *
+     * @param $id
+     * @return JsonResponse
+     *
+     * @Route(
+     *     "/api/v1/program/{id}",
+     *     name="program",
+     *     methods={"GET"},
+     *     requirements={"id"="\d+"}
+     * )
+     */
+    public function getProgram(int $id)
+    {
+        $program = $this->getDoctrine()
+            ->getRepository(Program::class)
+            ->find($id);
+
+        if ($program === null) {
+            return new JsonResponse("There is no academy with id of: " . $id, Response::HTTP_NOT_FOUND);
+        }
+
+        $reviewsArray = array();
+        foreach ($program->getReviews() as $review) {
+            $reviewsArray[] = $review->getReviewStars();
+        }
+
+        $startDatesArray = array();
+        $endDatesArray = array();
+        foreach ($program->getEvents() as $review) {
+            $startDatesArray[] = $review->getProgramStart();
+            $endDatesArray[] = $review->getProgramEnd();
+        }
+
+        return new JsonResponse(
+            array(
+                'program_id' => $program->getId(),
+                'program_name' => $program->getProgramName(),
+                'program_url' => $program->getProgramUrl(),
+                'program_price' => $program->getProgramPrice(),
+                'program_reviews' => $reviewsArray,
+                'program_start' => $startDatesArray,
```

Comment:
From this response, how I Will know when program starts and when the corresponding date is for that period?

Reply:
For the moment I cannot figure out how to get program end and start dates and especially program reviews from Academy controller.

Reply:
Why not creating nested array? Or even nested StdClass'es? 
They quite well transformed to json and used in front end.

Reply:
Are you talking about nested array of dates?

Reply:
Suggesting to give out json, which represents data structure similar to database:
`
{ "name": "academy_name", "email": "academy@email.com", "programs": [ { "name": "program1", "startDate": "2019-09-01", "endDate": "2020-02-01", "starRating": 4.6, "city": { "address": "some adress", "zip Code": "88654", "country": "netru" } } ] }
`

Reply:
@dimacellist will you implement this later?